src/tbp/monty/frameworks/models/evidence_matching/hypotheses_updater.py

```diff
@@ -74,6 +77,19 @@ def update_hypotheses(
         """
         ...
 
+    def remap_hypotheses_ids(
```

Comment:
Sorry that I'm coming back to this point again, but I just want us to get it right now to avoid confusion in the future. 

I think the name and docstring could still be clarified a bit further.

A name like `remap_hypotheses_ids_to_present` might better capture the arrow of time for the hypothesis ID remapping.

And just clarifying that the argument `hypotheses_ids` is from a previous time step, and the return is these updated to be meaningful in the current hypothesis space.

What do you think?